float32 startAngle
float32 endAngle
float32 avgDist